plugins/module_utils/vmware.py

```diff
@@ -218,3 +224,41 @@ def is_vcenter(self):
             return True
         elif api_type == 'HostAgent':
             return False
+
+    def __get_obj_by_name(self, vimtype, name):
+        """
+        Get the vsphere object associated with a given text name
+        Returns: Object if an object with the given type and name exists
+                 None otherwise
+        """
+        obj = None
+        container = self.content.viewManager.CreateContainerView(self.content.rootFolder, vimtype, True)
+        for c in container.view:
+            if c.name == name:
+                obj = c
+                break
+        return obj
+
+    def get_vm_by_name(self, vm_name, fail_on_missing=False):
```

Comment:
vm name is not unique. I think you should provide the root folder you're looking into
also, what is the strategy if there are multiple VMs with the same name?
For example, in community.vmware.vmware_guest_info there is the property `name_match`:
```
   name_match:
     description:
     - If multiple VMs matching the name, use the first or last found
     default: 'first'
     choices: ['first', 'last']
     type: str
```